src/statistics/timeSpent.ts

```diff
@@ -0,0 +1,104 @@
+import { Statistic } from "../objects/Statistic";
+
+const STATISTIC_KEY = "timePassed";
+const STATISTIC_TITLE = "Time Spent in";
+
+type TimeSpendStat = number;
+
+// A tad unwieldy, but we need to create a function that will be called whenever the statistic is updated
+// This function will be called every second, and will add 1 (for one second) to the existing time value
+const subscribeToTimePassing = (
+  // Create a callback function inside the main subscription function
+  // This callback function will be called in partnership with the accumulator function to update the stat value
+  updatedValueCallback: (addedTime: TimeSpendStat) => void
+) => {
+  // The hook in this case is time-based, so we want to call the callback function every second
+  context.setInterval(() => {
+    // In other widgets, we would probably use the value from this hook, but in this case we're just adding 1
+    updatedValueCallback(1);
+  }, 1000);
+};
+
+// each time the hook is called, we want to add to the existing time value
+// in this case, we know that the newVal will always be 1 since we're putting a static value in the updatedValueCallback
+// so we could replce the newVal with 1
+function accumulateSeconds(newVal: TimeSpendStat, existingVal: TimeSpendStat) {
+  return existingVal + newVal;
+}
+
+export const timeSpentStatistic = () => {
+  const key = STATISTIC_KEY;
+  const title = STATISTIC_TITLE;
+
+  const statistic = new Statistic({
+    key,
+    title,
+    resetValue: 0,
+    formatDisplay,
+    subscriber: subscribeToTimePassing,
+    accumulator: accumulateSeconds,
+  });
+
+  // LEGACY SUPPORT FOR TRACKING TIME IN-GAME
+  const legacyGameTimeKey = "openrct2-statistics.time.gameTime";
+  if (
+    // unfortunately there's no way to get rid of a key,
+    // so needing to check if it exists and if it has a value every time
+    context.sharedStorage.get(legacyGameTimeKey) != null
```

Comment:
When setting to `undefined`, then the following resolves to `false`:

```ts
context.sharedStorage.has(legacyGameTimeKey)
```

The key is deleted in the `plugin.store.json` file too. I'd imagine it's similar in a `.park` file, though I wouldn't be sure how to check since it's not a text file.

The comment can likely be removed and the code should be updated to the simpler expression. Same for legacy support for parks.